Source/WTF/wtf/CompactVariant.h

```diff
@@ -33,29 +33,33 @@ namespace WTF {
 // - Can only contain 254 or fewer alternatives.
 // - Has a more limited API, only offering `holds_alternative()` for type checking and `switchOn()` for value access.
 
-template<CompactVariantCapable V> class CompactVariant {
-    static_assert(std::variant_size_v<V> < 255);
-    using Variant = V;
+template<CompactVariantAlternative... Ts> class CompactVariant {
+    static_assert(sizeof...(Ts) < 255);
+    using StdVariant = std::variant<Ts...>;
     using Index = uint8_t;
     using Storage = uint64_t;
-    using Operations = CompactVariantOperations<V>;
+    using Operations = CompactVariantOperations<Ts...>;
 public:
     template<typename U> constexpr CompactVariant(U&& value)
-        requires std::constructible_from<V, U> && (!std::same_as<std::remove_cvref_t<U>, CompactVariant>)
+        requires
+            std::constructible_from<StdVariant, U>
+         && (!std::same_as<std::remove_cvref_t<U>, CompactVariant>)
+         && (!IsStdInPlaceTypeV<std::remove_cvref_t<U>>)
+         && (!IsStdInPlaceIndexV<std::remove_cvref_t<U>>)
```

Comment:
Why the parentheses?

Reply:
Weird requirement of the requires syntax.